integration/csv_fork_test.go

```diff
@@ -0,0 +1,687 @@
+// Copyright (c) 2016 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+// This file is ignored during the regular tests due to the following build tag.
+// +build rpctest
+
+package integration
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/rpctest"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+)
+
+// makeTestOutput creates an on-chain output paying to a freshly generated
+// p2pkh output with the specified amount.
+func makeTestOutput(r *rpctest.Harness, t *testing.T,
+	amt btcutil.Amount) (*btcec.PrivateKey, *wire.OutPoint, []byte, error) {
+
+	// Create a fresh key, then send some coins to an address spendable by
+	// that key.
+	key, err := btcec.NewPrivateKey(btcec.S256())
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// Using the key created above, generate a pkScript which it's able to
+	// spend.
+	a, err := btcutil.NewAddressPubKey(key.PubKey().SerializeCompressed(), r.ActiveNet)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	selfAddrScript, err := txscript.PayToAddrScript(a.AddressPubKeyHash())
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	output := &wire.TxOut{PkScript: selfAddrScript, Value: 1e8}
+
+	// Next, create and broadcast a transaction paying to the output.
+	fundTx, err := r.CreateTransaction([]*wire.TxOut{output}, 10)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	txid, err := r.Node.SendRawTransaction(fundTx, true)
```

Comment:
Please make all of these `txid` vars `txHash` instead.  We have been very careful to avoid mischaracterizing the hash as an ID in general.  Improperly treating the transaction hash as a unique ID is one of the main reason malleability led to so many issues in various software.  I do realize a big reason for segwit is stable hashes so they can be (ab)used as an ID, but I'd really prefer to avoid the notion altogether in the code since non-segwit transactions still exist and it applies there even less.